### DISCUSSION TITLE
Ublox 9 support

### DISCUSSION THREAD

**Discussion by gwbres** [ublox-rs/ublox#72] [Q&A]
@andrei-ng have you ever had a chance to test this library against M9/F9 devices ?

I'm not sure how the library behaves against latest receivers.
And I'm thinking of upgrading to an F9 device for my current developments

**Comment by andrei-ng** [comment]
@gwbres , I only ran it once against an F9R device and it seemd to work fine. However, it was a limited test, and I never came back to it.

I also want to switch to using F9R instead of my M8L device for my developments so we can share the workload + testing.

**Reply by gwbres** [reply]
@andrei-ng awesome, thank you for the feedback.

I am not familiar with the upgrades in the protocol, are you ? (trying to estimate the workload).
We might have to introduce a crate feature if the protocols differ too much, anyways we will have to decide first what the best approach is. 

So far I'll just test and get more familiar with the new device family and will keep you updated. 
I will put my hands on an F9P device by next january. My usage is:

* RINEX post processing with my `rinex-cli` apps
* CGGTTS with my `rnx2cggtts` apps
* CGGTTS formatting with my `ubx2cggtts` apps
* RINEX formatting with my `ubx2rnx` apps

**Reply by andrei-ng** [reply]
Nope, I did not look into the protocol changes either. I will probably look into this beginning of next year.

Yes, probably this will make us tackle #43 .

My use cases are a bit different, more towards robotics/navigation: RTK and ADR usage. As we have different use cases, which seem orthogonal, I think all would give a nice coverage of scenarios.

**Reply by gwbres** [reply]
Hello @andrei-ng 

FYI, I will rapidly get back to applications that involve hardware (Jan 2024 worst case scenario).  
I am in the process of getting access to an M8T and F9T devkits, which will help future developments

**Comment by andrei-ng** [comment]
Hi @gwbres ,
Time to get back to this thread as well :).
I have been using an F9R device with this crate for the past 6 months. I haven't used many messages besides NAV-PVT, MON-VER and ESF messages. All has worked fine so far. 

I did notice that the new protocol specification is going in the direction of deprecating some specific configuration messages in favor of CfgValSet and CfgValGet. So I expecting that soon support for CFG specific messages will go away all together.

**Reply by gwbres** [reply]
> I did notice that the new protocol specification is going in the direction of deprecating some specific configuration messages in favor of CfgValSet and CfgValGet

Yeah that's what you said this morning, I was not aware of that. Our CfgVal support is quite good already so we're fine (expect maybe the pending CfgValGet)

> So I expecting that soon support for CFG specific messages will go away all together

well we should preserve support for proto23 that we have today

> I have been using an F9R device with this crate for the past 6 months. I haven't used many messages besides NAV-PVT, MON-VER and ESF messages. All has worked fine so far.

I'll have the same usage, I expect that this will work

**Reply by andrei-ng** [reply]
> well we should preserve support for proto23 that we have today

Of course.
